identity-account-storage/src/storage/traits.rs

```diff
@@ -118,6 +120,31 @@ pub trait Storage: storage_sub_trait::StorageSendSyncMaybe + Debug {
   /// Returns `true` if a key exists at the specified `location`.
   async fn key_exists(&self, did: &IotaDID, location: &KeyLocation) -> Result<bool>;
 
+  /// Encrypts the given `data` with the specified `algorithm`
+  ///
+  /// Diffie-Helman key exchange will be performed in case an [`KeyType::X25519`] is given.
+  async fn encrypt_data(
```

Comment:
Since this sets the values of the ephemeral keys, we can't add it to the StorageTestSuite. I added a test for the stronghold type, but I feel it doesn't add much more than what we current assert in the `encryption` example.

Reply:
I think it's still valuable to have this test, because it proves (a large chunk of) our implementation is compatible with other implementations. With an extended storage test suite we can additionally ensure a working round-trip within our implementation.